docker/bitcoind/Dockerfile

```
@@ -1,19 +1,36 @@
-# Use the official Ubuntu 22.04 base image
+# Use the official Ubuntu 24.04 base image
 FROM ubuntu:24.04
 
 WORKDIR /app
 
-# Install necessary dependencies and add PPA
+# Install necessary dependencies
 RUN apt-get update && \
-    apt-get install -y software-properties-common && \
-    add-apt-repository ppa:luke-jr/bitcoincore && \
-    apt-get update && \
-    apt-get install -y \
-    bitcoind \
-    curl && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
+    apt-get install -y software-properties-common build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 && \
+    apt-get install -y libevent-dev libboost-dev libsqlite3-dev && \
+    apt-get install -y libssl-dev libboost-all-dev git
+
+# Set Git with dummy user configuration
+# This configuration is needed for cherry-picking the specific commit
+RUN git config --global user.email "john@example.com" && \
+    git config --global user.name "John Doe"
+
+# Clone Bitcoin repository and checkout the commit for v28.0 tag
+RUN git clone https://github.com/bitcoin/bitcoin.git && \
+    cd bitcoin && \
+    git checkout 5de225f5c145368f70cb5f870933bcf9df6b92c8
+
```

Yes @bewakes is right, use https://github.com/bitcoin/bitcoin/releases/tag/v28.0
```suggestion
RUN git clone -b 'v28.0' --single-branch --depth 1 https://github.com/bitcoin/bitcoin.git && 
    cd bitcoin
```
Also you don't need to pull all the ~crap~ things from the repo, you can do `--single-branch` and `--depth 1`

> [!TIP]
> You might want to make `v28.0` configurable.

---

Source: https://stackoverflow.com/a/31666461